src/FormParserFactory.php

```diff
@@ -0,0 +1,37 @@
+<?php
+
+namespace React\Http;
+
+use React\Http\Parser\FormUrlencoded;
+use React\Http\Parser\Multipart;
+use React\Http\Parser\NoBody;
+use React\Http\Parser\ParserInterface;
+
+class FormParserFactory
+{
+    /**
+     * @param Request $request
+     * @return ParserInterface
+     */
+    public static function create(Request $request)
+    {
+        $headers = $request->getHeaders();
+        $headers = array_change_key_case($headers, CASE_LOWER);
+
+        if (!array_key_exists('content-type', $headers)) {
+            return new NoBody($request);
+        }
+
+        $contentType = strtolower($headers['content-type']);
+
+        if (strpos($contentType, 'multipart/') === 0) {
+            return new Multipart($request);
+        }
+
+        if (strpos($contentType, 'application/x-www-form-urlencoded') === 0) {
+            return new FormUrlencoded($request);
+        }
+
+        return new NoBody($request);
```

Comment:
Cool!

I don't understand your last point. I actually only want a `RawBody` if the incoming request has either

1. No Content-Type
2. A Content-Type reactphp/http can not automatically decode (means no `multipart/*', `application/x-www-form-urlencoded`)

`NoBody` should only be used when no Content-Length is given (because we can not detect the end of the request body).

I don't understand what a streamed request has to do with the RawBody. :) What do I get when second parameter is `false` and there's no content-type (or one that we do not decode)?